### PR TITLE
🐛  Check to see if obj is nil so there is no panic trying to get the gvk

### PR DESCRIPTION
**Pull request by emmjohnson**
If object is nil, this func will panic.

Co-authored-by: Marty Spiewak <mspiewak@vmware.com>

### CODE REVIEW

File: pkg/log/zap/zap_test.go [hunk 1]
```diff
@@ -255,6 +255,15 @@ var _ = Describe("Zap logger setup", func() {
 					"namespace": name.Namespace,
 				}))
 			})
+
+			It("should not panic with nil obj", func() {
+				pod := &corev1.Pod{}
```
Comment (alvaroaleman) [inline pkg/log/zap/zap_test.go:260]: Nit to get a typed nilpointer:
```suggestion
				var pod *corev1.Pod
```

**Review by vincepri — APPROVED** [review]
/ok-to-test
/lgtm

nice find!

**Comment by alvaroaleman** [comment]
/test pull-controller-runtime-apidiff-master
Not sure why the job doesn't actually start :thinking:

**Comment by alvaroaleman** [comment]
Actually its optional anyways
/lgtm
/approve

Thanks!

**Pull request merged**